test/automate/automate-utils.js

```diff
@@ -14,285 +14,292 @@
-         base64Data,
-         'base64',
-         function(err) {
-          if (err) { console.log(err); }
-         }
-       );
-     }
-   });
- };
-
-// Chromedriver is used by Selenium and requiring the module avoids having to
-// download and install the executable from Google along with setting the path.
-/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "chromedriver" }] */
+// Chromedriver and geckodriver are used by Selenium and requiring the module
+// avoids having to download and install the executable from Google
+// along with setting the path.
+/* eslint no-unused-vars: ["error", {
+    "varsIgnorePattern": "chromedriver|geckodriver"
+   }] */
 let chromedriver = require('chromedriver');
+let geckodriver = require('geckodriver');
 
 let webdriver = require('selenium-webdriver'),
     By = webdriver.By,
     until = webdriver.until;
 
-// Used to save screenshots.
-const fs = require('fs');
-const path = require('path');
-
-exports.doesProjectPassTests = function(name, URL) {
+exports.doesProjectPassTests = (browser, name, URL) => {
 
   // If elements don't appear within 1 minute, it's usually an error.
   const elementTimeout = 60000;
 
   // If all of the project tests pass, this is set to true.
   let success = false;
 
-  const chrome = require('selenium-webdriver/chrome');
+  // headless mode for chrome and firefox is enabled by default
+  let headless = true;
+  if (typeof process.env.HEADLESS !== 'undefined') {
+    // eslint-disable-next-line eqeqeq
+    headless = process.env.HEADLESS == true;
+  }
+
+  let capabilities;
+
+  if (browser === 'chrome') {
+
+    // Set up Chrome options.
+    const chrome = require('selenium-webdriver/chrome');
+    let chromeOptions = (new chrome.Options())
+      .addArguments([
+        'log-level=3',
+        'mute-audio',
+        'no-sandbox'
+      ])
+      .setChromeBinaryPath(chromeBinaryPath);
+    if (headless) {
+      chromeOptions.headless();
+    }
 
-  // Set up Chrome options. We use both "start-maximized" along with
-  // "window-size" because some platforms prefer one or the other.
-  let options = new chrome.Options();
-  options.addArguments([
-    'start-maximized',
-    `window-size=${browserMaxWidth}x${browserMaxHeight}`,
-    'no-sandbox'
-  ]);
-  options.setChromeBinaryPath(chromeBinaryPath);
+    capabilities = chromeOptions.toCapabilities();
+
+  } else if (browser === 'firefox') {
+
+    // Set up Firefox options.
+    const firefox = require('selenium-webdriver/firefox');
+    let profile = new firefox.Profile();
+    profile.setPreference('media.volume_scale', '0.0');
+    let firefoxOptions = (new firefox.Options())
+      .setProfile(profile)
+      .setBinary(firefoxBinaryPath);
+    if (headless) {
+      firefoxOptions.headless();
+    }
+
+    capabilities = firefoxOptions.toCapabilities();
+
+  } else {
+    throw new Error(`Browser ${browser} is not supported`);
+  }
+
+  capabilities.set('acceptInsecureCerts', true);
```

Comment:
Why are we setting this? Could you document what happens if we don't have this line?

Reply:
documented